eng/Versions.props

```diff
@@ -7,6 +7,8 @@
     <!-- Always use shipping version instead of dummy version -->
     <DotNetUseShippingVersions>true</DotNetUseShippingVersions>
     <PreReleaseVersionLabel>preview</PreReleaseVersionLabel>
+    <!-- Opt-in repo features -->
+    <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
```

Comment:
Do we still need this?  It's controlling the PackageReferences which it looks like you've handled by your changes in the tools.props.
https://github.com/dotnet/arcade/blob/b5f99257d63e4f7784412ef78c5ca11ffea75338/src/Microsoft.DotNet.Arcade.Sdk/tools/Settings.props
https://github.com/dotnet/arcade/blob/b5f99257d63e4f7784412ef78c5ca11ffea75338/src/Microsoft.DotNet.Arcade.Sdk/tools/Compiler.props

My initial recommendation was to only set this property and ensure that tools.props would get the compiler.props file imported.

Reply:
We might not, but it also does no-harm in having it and will, if nothing else, make it simpler to support these toolsets properly if we ever start doing per-project restore.